projects/saturn/src/main/java/io/fairspace/saturn/services/views/ViewStoreClientFactory.java

```diff
@@ -103,62 +102,103 @@ void ensureTableExists(Table table) throws SQLException {
+            var resultSet = connection.getMetaData().getTables(null, null, table.name, null);
+            var tableExists = resultSet.next();
+            
+            if (!tableExists) {
+                createTable(table, connection);
+                createIndexes(table, connection);
+            } else {
+                updateTable(table, connection);
+            }
+        }
+    }
+
+    private void updateTable(Table table, Connection connection) throws SQLException {
+        var columnMetadata = getColumnMetadata(connection, table.name);
+        var newColumns = table.columns.stream()
+                .filter(column -> !column.type.isSet() && !columnMetadata.containsKey(column.name))
+                .collect(Collectors.toList());
+        try {
+            log.debug("New columns: {}", new ObjectMapper().writeValueAsString(newColumns));
+        } catch (JsonProcessingException e) {
+            e.printStackTrace();
+        }
+        // Update existing table
+        if (!newColumns.isEmpty()) {
+            connection.setAutoCommit(true);
+            var command = newColumns.stream()
+                    .map(column -> String.format("alter table %s add column %s %s",
+                            table.name, column.name, databaseTypeForColumnType(column.type)))
+                    .collect(Collectors.joining("; "));
+            log.debug(command);
+            connection.createStatement().execute(command);
+            connection.setAutoCommit(false);
+            log.info("Table {} updated.", table.name);
+        }
+    }
+
+    private void createTable(Table table, Connection connection) throws SQLException {
+        // Create new table
+        connection.setAutoCommit(true);
+        var columnSpecification = table.columns.stream()
+                .filter(column -> !column.type.isSet())
+                .map(column -> String.format("%s %s", column.name, databaseTypeForColumnType(column.type)))
+                .collect(Collectors.joining(", "));
+        var keys = table.columns.stream()
+                .filter(column -> column.type == ColumnType.Identifier)
+                .map(column -> column.name)
+                .collect(Collectors.joining(", "));
+        var command = String.format("create table %s ( %s, primary key ( %s ) )", table.name, columnSpecification,
+                keys);
+        log.debug(command);
+        connection.createStatement().execute(command);
+        connection.setAutoCommit(false);
+        log.info("Table {} created.", table.name);
+    }
+
+    private void createIndexes(Table table, Connection connection) throws SQLException {
+        
+        var keys = table.columns.stream()
+                .filter(column -> column.type == ColumnType.Identifier)
+                .map(column -> column.name)
+                .toList();
+
+        connection.setAutoCommit(true);
+
+        for (var column : keys) {
+            var indexName = String.format("%1$s_%2$s_idx", table.name, column);
```

Comment:
`%1$s_%2$s_idx` could be replaced with  just`%s_%s_idx`

Reply:
yes that's right, I'll change